### PR TITLE
Software Testing Infrastructure

### PR DESCRIPTION
**Pull request by HU90m**
This moves the testing infrastructure from `sonata-software` (https://github.com/lowRISC/sonata-software/pull/26) to this repository. The tests have been adapted to run in the minimal bare metal environment.

What was in the tests directory is now in a directory called 'checks'. The distinction is talked about in `sw/cheri/README.md`.

### CODE REVIEW

File: sw/cheri/tests/xmake.lua
Comment (engdoreis) [inline sw/cheri/tests/xmake.lua:1]: remove this file

**Review by engdoreis — APPROVED** [review]
LGTM. But it would be nice if Gary or Harry could review the Nix code.

File: util/test_runner.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+                self.simulator_binary: Path = Path(
+                    str(binary)
+                    if (binary := shutil.which("sonata-simulator"))
+                    else "./sonata-simulator"
+                )
+            if not self.sim_boot_stub:
+                if stub := os.getenv("SONATA_SIM_BOOT_STUB"):
+                    self.sim_boot_stub: Path = Path(stub)
+                else:
+                    print("No simulator boot stub found")
```
[the remark this answers is not in the document]
Comment (HU90m) [reply util/test_runner.py]: We could in the future

File: sw/cheri/tests/boot.S [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+	.globl start
+	.p2align 2
+    .type start,@function
+start:
+	// ca0 (first argument) contains the read-write root
+	cspecialr        ca0, mtdc
+
+	// No bounds on stack, grows down from the end
+	li               sp, 0x0010E000
```
[the remark this answers is not in the document]
Comment (HU90m) [reply sw/cheri/tests/boot.S:14]: That's a nice idea

**Pull request merged**